### PR TITLE
New Beta Resource: `azurerm_app_service_source_control_slot` 

### PR DESCRIPTION
**Pull request by jackofallops**
Also addresses some issues in `azurerm_app_service_source_control` while passing through.

### CODE REVIEW

**Comment by jackofallops** [comment]
Tests looking good:

image

File: internal/services/appservice/source_control_slot_resource.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+			}
+
+			props := *appSourceControl.SiteSourceControlProperties
+
+			state := SourceControlSlotModel{
+				SlotID:                    id.ID(),
+				SCMType:                   string(siteConfig.ScmType),
+				RepoURL:                   utils.NormalizeNilableString(props.RepoURL),
+				Branch:                    utils.NormalizeNilableString(props.Branch),
+				ManualIntegration:         *props.IsManualIntegration,
```
Comment (mbfrahry) [inline internal/services/appservice/source_control_slot_resource.go:261]: Any chance of these and the next three lines being nil?

File: website/docs/r/app_service_source_control_slot.html.markdown [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+* `use_mercurial` - (Optional) The repository specified is Mercurial. Defaults to `false`. Changing this forces a new resource to be created.
+
+---
+
+A `github_action_configuration` block supports the following:
+
+* `code_configuration` - (Optional) A `code_configuration` block as detailed below.
+
+* `container_configuration` - (Optional) A `container_configuration` block as detailed below.
```
Comment (mbfrahry) [inline website/docs/r/app_service_source_control_slot.html.markdown:87]: Could we add a test for this block?

**Review by mbfrahry — APPROVED** [review]
LGTM with a few minor things to add in addition to some linting fixes

File: internal/services/appservice/source_control_slot_resource.go (hunk 1 above)
Comment (jackofallops) [reply internal/services/appservice/source_control_slot_resource.go:261]: oh, good spot, thought I'd caught all those!

File: website/docs/r/app_service_source_control_slot.html.markdown (hunk 2 above)
Comment (jackofallops) [reply website/docs/r/app_service_source_control_slot.html.markdown:87]: Unfortunately, not meaningfully, no. It requires write access to a GitHub repository to create a GHA / Workflow file(s). It _appears_ to work as far as I could take it manually, but it's not something that can be reasonably done from the AccTests.

**Comment by jackofallops** [comment]
Testing re-run for expanded scope of change, looking good still.

image

**Pull request merged**